whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapServiceTestIntegration.java

```diff
@@ -2283,6 +2295,108 @@ public void lookup_org_entity() {
+                "tech-c:        TP2-TEST\n" + //has email and notify
+                "mnt-ref:       OWNER-MNT\n" +
+                "mnt-ref:       ANOTHER-MNT\n" +
+                "mbrs-by-ref:   OWNER-MNT, ANOTHER-MNT\n" +
+                "mnt-lower:       ANOTHER-MNT\n" +
+                "mnt-domains:       ANOTHER-MNT\n" +
+                "mnt-routes:       ANOTHER-MNT\n" +
+                "mnt-by:        ANOTHER-MNT\n" +
+                "created:         2011-07-28T00:35:42Z\n" +
+                "last-modified:   2019-02-28T10:14:46Z\n" +
+                "source:        TEST");
+
+        final Entity entity = createResource("entity/ORG-ONE-TEST")
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(Entity.class);
+
+        assertThat(entity.getRedacted().size(), is(9));
+
+        assertThat(entity.getRedacted().get(0).getName().getDescription(), is("Updates notification e-mail information"));
+        assertThat(entity.getRedacted().get(0).getReason().getDescription(), is("Personal data"));
+        assertDoesNotThrow(() -> JsonPath.compile(entity.getRedacted().get(0).getPrePath()));
+        assertThat("$.entities[?(@.roles=='registrant')].vcardArray[1][?(@[0]=='notify')]",
+                is(entity.getRedacted().get(0).getPrePath()));
+        assertThat(entity.getRedacted().get(0).getMethod(), is("removal"));
+
+        assertThat(entity.getRedacted().get(1).getName().getDescription(), is("Updates notification e-mail information"));
+        assertThat(entity.getRedacted().get(1).getReason().getDescription(), is("Personal data"));
+        JsonPath jsonPath = assertDoesNotThrow(() -> JsonPath.compile(entity.getRedacted().get(1).getPrePath()));
+
+        assertThat("$.entities[?(@.roles=='technical && administrative')].vcardArray[1][?(@[0]=='notify')]", is(entity.getRedacted().get(1).getPrePath()));
+        assertThat(entity.getRedacted().get(1).getMethod(), is("removal"));
+
+        assertThat(entity.getRedacted().get(2).getName().getDescription(), is("Authenticate more specific resources"));
+        assertThat(entity.getRedacted().get(2).getReason().getDescription(), is("No registrant mntner"));
+        assertDoesNotThrow(() -> JsonPath.compile(entity.getRedacted().get(2).getPrePath()));
+        assertThat("$.entities[?(@.handle=='ANOTHER-MNT')]", is(entity.getRedacted().get(2).getPrePath()));
+        assertThat(entity.getRedacted().get(2).getMethod(), is("removal"));
+
+        assertThat(entity.getRedacted().get(3).getName().getDescription(), is("Authenticate incoming references"));
+        assertThat(entity.getRedacted().get(3).getReason().getDescription(), is("No registrant mntner"));
+        assertDoesNotThrow(() -> JsonPath.compile(entity.getRedacted().get(3).getPrePath()));
+        assertThat("$.entities[?(@.handle=='OWNER-MNT')]", is(entity.getRedacted().get(3).getPrePath()));
+        assertThat(entity.getRedacted().get(3).getMethod(), is("removal"));
```

Comment:
can you create a method for assertations and pass on entity.getRedacted().get(index) ?

Reply:
Done, looks much better and with less duplicated code. Thanks!